plugin.go

```diff
@@ -81,7 +83,22 @@ func (p *Plugin) Exec() error {
 		awsConfig.Credentials = credentials.NewStaticCredentials(p.Key, p.Secret, "")
 	}
 	awsConfig.Region = aws.String(p.Region)
-	svc := ecs.New(session.New(&awsConfig))
+
+	var svc *ecs.ECS
+	sess := session.Must(session.NewSession(&awsConfig))
+
+	// If user role ARN is set then assume role here
+	if len(p.UserRoleArn) > 0 {
+		var arnCredentials *credentials.Credentials
+
+		awsConfigArn := aws.Config{Region: aws.String(p.Region)}
+		arnCredentials = stscreds.NewCredentials(sess, p.UserRoleArn, func(p *stscreds.AssumeRoleProvider) {})
```

Comment:
I think this can just be ```arnCredentials := stscreds.NewCredentials(sess, p.UserRoleArn)``` and you can delete line 92 but if you don't mind trying that out :)